realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -162,18 +162,22 @@ static void finalize_object(jlong ptr)
 
 static inline size_t do_create_row(jlong shared_realm_ptr, jlong table_ptr)
 {
-    auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     auto& table = *(reinterpret_cast<realm::Table*>(table_ptr));
-    shared_realm->verify_in_write();
+    if (shared_realm_ptr) {
```

Comment:
right ... but this function could be on a very hot path (the bulk insertion for example.), add the check to support test cases may not worth it.

I think i have the same problem before, what I did is create a `SharedRealm` for the relevant JNI tests before.

IMO, some JNI tests could be just removed, like the `PivotTest.java`, some should be moved to upper layer tests. For those which are difficult to move upper, keep it in the JNI tests and create a `SharedRealm` instead. Support `Table` without `SharedRealm` doesn't have too much value for java binding.